src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Metadata.cs

```diff
@@ -85,194 +96,435 @@ private static Metadata GetMetadataNoThrow(PortableExecutableReference reference
-        }
 
-        private static void GenerateMetadataNodes(
-            MetadataReader reader,
-            string name,
-            int parentIndex,
-            OrderPreservingMultiDictionary<string, MetadataDefinition>.ValueSet definitionsWithSameName,
-            List<Node> unsortedNodes)
-        {
-            var node = new Node(name, parentIndex);
-            var nodeIndex = unsortedNodes.Count;
-            unsortedNodes.Add(node);
+            private void LookupMetadataDefinitions(
+                TypeDefinition typeDefinition,
+                OrderPreservingMultiDictionary<string, MetadataDefinition> definitionMap)
+            {
+                // Only bother looking for extension methods in static types.
+                if ((typeDefinition.Attributes & TypeAttributes.Abstract) != 0 &&
+                    (typeDefinition.Attributes & TypeAttributes.Sealed) != 0)
+                {
+                    foreach (var child in typeDefinition.GetMethods())
+                    {
+                        var method = _metadataReader.GetMethodDefinition(child);
+                        if ((method.Attributes & MethodAttributes.SpecialName) != 0 ||
+                            (method.Attributes & MethodAttributes.RTSpecialName) != 0)
+                        {
+                            continue;
+                        }
+
+                        // SymbolTreeInfo is only searched for types and extension methods.
+                        // So we don't want to pull in all methods here.  As a simple approximation
+                        // we just pull in methods that have attributes on them.
+                        if ((method.Attributes & MethodAttributes.MemberAccessMask) == MethodAttributes.Public &&
+                            (method.Attributes & MethodAttributes.Static) != 0 &&
+                            method.GetCustomAttributes().Count > 0)
+                        {
+                            var definition = new MetadataDefinition(
+                                MetadataDefinitionKind.Member, _metadataReader.GetString(method.Name));
+
+                            definitionMap.Add(definition.Name, definition);
+                        }
+                    }
+                }
 
-            // Add all child members
-            var definitionMap = OrderPreservingMultiDictionary<string, MetadataDefinition>.GetInstance();
-            try
+                foreach (var child in typeDefinition.GetNestedTypes())
+                {
+                    var type = _metadataReader.GetTypeDefinition(child);
+
+                    // We don't include internals from metadata assemblies.  It's less likely that
+                    // a project would have IVT to it and so it helps us save on memory.  It also
+                    // means we can avoid loading lots and lots of obfuscated code in the case the
+                    // dll was obfuscated.
+                    if (IsPublic(type.Attributes))
```

Comment:
OK, that's what I thought. I just wanted to make sure I understood the scenario--I agree it would be very weird to use a metadata assembly in such a way that this would even matter.